.circleci/main.yml

```diff
@@ -542,6 +550,45 @@ jobs:
       - store_artifacts:
           path: /tmp/debug-dump
           destination: debug-dump
+  create-nightly-tag:
+    executor: docker-go
+    steps:
+      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - "4c:1c:ff:56:c1:d0:07:03:23:08:08:2e:e6:16:30:1d"
+      - run:
+          name: setup pachydermbuildbot git
+          command: |
+            git config --global user.email "buildbot@pachyderm.io"
+            git config --global user.name "pachydermbuildbot"
+      - run:
+          name: create nightly tag if new commits
+          command: |
+            changes=$( env -i git log --pretty=oneline --since='yesterday' | wc -l )
+            if [ $changes -gt 0 ]; then
+              TIMESTAMP=$(date +'%Y%m%d%T' | sed 's/://g')
+              WORKINGVERSION=<< pipeline.parameters.working-pachyderm-version >>
+              NIGHTLY=${WORKINGVERSION}-nightly.${TIMESTAMP}
+              git tag -f -am "nightly release tag ${NIGHTLY}" ${NIGHTLY}
+              git push -f origin ${NIGHTLY}
```

Comment:
I don't think we should push here, and especially not force push.  I think it should just be `git push --tags`.  If it already exists, the process should fail.  If code changes, that seems non-ideal as well.  It's certainly not worth overwriting commits with whatever we have here, which is what "git push -f" does.

Reply:
good idea. using ` git push origin ${NIGHTLY} `now